### PR TITLE
feat(returns): value net units at market, not lot-matched inventory

### PR DESCRIPTION
**Pull request by robcohen**
## Summary

Re-architects the returns engine to **value net units at market** instead of realizing holdings through the lot-matching `BookingEngine`. This is the core of the reframed #1850 (§1), and it reverses the strict-clean broken-ledger behavior shipped in #1849.

Money-weighted (XIRR) and time-weighted returns depend only on cash flows and terminal **market** value (`net units × price`) — never on cost-basis lots. So the engine now sums each account's complete posting units per commodity **without lot-matching** and values the net at market.

### Why (the research reversal)

#1849 shipped strict-clean: any booking error → clean `Err`. Research (reading beangrow's `returns.py` + beancount mailing-list threads) showed that was wrong — the leniency it blocked is a genuinely useful, common capability. An over-sell, or an empty-cost `{}` sale with no matching lot, is the **normal state of imported brokerage data**; beangrow sums `Inventory.add_position` (no lot-matching) and values at market, so it never crashes. rledger's fragility was self-inflicted: it valued net units at market but *sourced* them from the full lot-matching engine, which traps on unmatched reductions — lot-matching that market valuation never needed.

### Behavior

- **Cost-basis/lot errors are now a non-event for the report** — an over-sell or empty-cost `{}` sale simply nets the units (possibly negative) and values at market, like beancount + beangrow. `rledger check` remains the validator.
- **Only genuinely-unvaluable input errors** — an in-scope account with an elided/uninterpolated posting (net units unknown) → `ExtractError::UnbookedInput`.
- **Clean-ledger output is byte-identical** — the existing per-date / per-scope drift guards stay green.

## Changes

- **`rustledger-returns/src/extract.rs`**: a `NetUnits` accumulator (`BTreeMap<Account, BTreeMap<Currency, Decimal>>` + an `unvaluable` set) replaces the `BookingEngine` in `investment_value_at` / `investment_values_at` / `investment_values_multi`. The accumulation pass can no longer error; valuation errors move per-date / per-scope — so `UnbookedInput` is now **per-scope** in the `--by-group` path (unblocks §4 partial render).
- **Dependency cleanup**: drop `rustledger-booking` from `rustledger-returns`; remove the now-dead `BookingEngine::try_apply` from `rustledger-booking` (§3).
- **Tests**: rewrite the three #1849 strict-clean tests (returns, CLI, FFI component) to assert the tolerant net-units behavior — an over-sell nets negative and values at market; only an elided in-scope posting errors.
- **Docs**: module + function docs updated from the booking-realization framing to net-units.

## Verification

- `rustledger-returns` 62 · `rustledger-booking` 110+ · `rustledger-ffi-component` 30 · `rustledger` CLI 424 — all pass
- `cargo check --workspace --all-features --all-targets`, `clippy` @1.97.0, `rustdoc -D warnings` — all clean
- pre-push hooks (rustfmt, typos, gitleaks, workspace check, unsafe-invariant, plugin-test-quality) — all pass

## Follow-ups (remaining #1850 increments)

§2 report-vs-validator separation · §4 `--by-group` CLI partial-render (now unblocked) · §5 edge cases (falling-price / negative-return tests, missing-price policy) · §6 messy-imported-ledger + beangrow cross-check suite · §7 docs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/rustledger-booking/src/book.rs [hunk 1]
```diff
@@ -579,51 +579,26 @@ impl BookingEngine {
     /// matching lot yet is dropped (its `reduce` error is ignored, historical
     /// release behavior). The loader pipeline guarantees this ordering; the
     /// in-loop `debug_assert` below catches a violating caller in debug builds.
-    /// A caller that CANNOT guarantee booked input (e.g. valuing a returns scope
-    /// over a ledger whose loader re-merged a booking-failed transaction) must
-    /// use [`Self::try_apply`] instead, which surfaces the failure as an `Err`.
+    /// A caller that CANNOT guarantee booked input must not derive a figure from
+    /// this lot-matching realization — e.g. the returns engine values **net units
+    /// at market** (`rustledger-returns`), never `apply`, so a re-merged
+    /// booking-failed transaction cannot over-sell it.
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-booking/src/book.rs:585]: try_apply was added in #1849 (unreleased) solely for this realization path, whose only consumer is removed here. No external users, and rustledger-booking is 0.x, so no deprecated wrapper or extra version bump is needed.

File: crates/rustledger-returns/src/extract.rs [hunk 2]
```diff
@@ -420,109 +435,127 @@ pub fn terminal_value(
...
+    // net units — the one shape net-units cannot value — so error rather than
+    // understate it. Scanned over the whole `unvaluable` set (a `BTreeSet`, so the
+    // choice of named account is deterministic), including accounts that appear
+    // ONLY with an elided leg and so never entered `per_account`.
+    for account in &holdings.unvaluable {
+        if scope.classify(account.as_str()) == AccountRole::Investment {
+            return Err(ExtractError::UnbookedInput(format!(
+                "account {account} has an un-booked (elided) posting; cannot value returns"
+            )));
+        }
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-returns/src/extract.rs:547]: Added compute_returns_multi_isolates_an_unvaluable_scope in 404674d5 — one scope errors UnbookedInput while a disjoint scope stays Ok, over one shared accumulation.

File: crates/rustledger-booking/src/book.rs [hunk 3]
```diff
@@ -579,9 +579,10 @@ impl BookingEngine {
...
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-booking/src/book.rs:585]: Fixed — the debug_assert message no longer references the removed try_apply; it now points to the net-units path.

File: crates/rustledger-returns/src/extract.rs [hunk 4]
```diff
@@ -336,11 +346,11 @@ pub fn extract_flows(
...
-            // a contract violation the leaf crate cannot detect, exactly as
-            // `apply`/`account_balances` also require pre-booked input).
+            // The input-contract interpolated stream (module docs) fills units on
+            // every posting, so a None here is unreachable in practice; skipping is
+            // defensive, not a silent drop of a real flow (an un-interpolated
+            // stream is a contract violation the leaf crate cannot detect, exactly
+            // as `account_balances` also requires interpolated input).
             let Some(amount) = posting.amount() else {
                 continue;
             };
```
[the remark this answers is not in the document]
Comment (robcohen) [reply crates/rustledger-returns/src/extract.rs]: Fixed in this PR — extract_flows now returns Err(UnbookedInput) on an elided external leg instead of dropping it (see the elided_external_leg_errors_not_dropped test).
